### DISCUSSION TITLE
Did PowerShell just crash?

### DISCUSSION THREAD

**Discussion by beedell-roke** [cloudbase/powershell-yaml#178] [Q&A] [answered]
Please see https://github.com/cloudbase/powershell-yaml/discussions/178#discussioncomment-11649645. It replaces this.

**Comment by gabriel-samfira** [comment]
Could you include the OS name and version, PowerShell version and powershell-yaml version? I will need yo reproduce.

**Reply by gabriel-samfira** [reply]
This is on a Windows Server 2019 with the latest powershell-yaml (0.4.11):

![image](https://github.com/user-attachments/assets/81143eac-33c2-4535-baed-c86476f6ceeb)

**Reply by gabriel-samfira** [reply]
I can't seem to reproduce this.

**Reply by beedell-roke** [reply]
> https://github.com/cloudbase/powershell-yaml/discussions/178#discussioncomment-11648501

@gabriel-samfira, I actually didn't mean to ask this here. Really sorry. Meant to create a discussion at the main PowerShell repository! XD I obviously didn't check the URI.

I don't have any reason to think it's caused by PS-YAML, unless you do (the dump is fairly meaningless to me, but perhaps not you)?

**Reply by beedell-roke** [reply]
> https://github.com/cloudbase/powershell-yaml/discussions/178#discussioncomment-11647451

@gabriel-samfira, I'll provide them when I'm next on my PC, if they're still of use.

**Reply by gabriel-samfira** [reply]
> I don't have any reason to think it's caused by PS-YAML, unless you do (the dump is fairly meaningless to me, but perhaps not you)?

Not really. I was a bit confused. Initially I thought this may have been caused by the way powershell-yaml loads its assemblies now, but it's not. In any case, good to know.

**Reply by beedell-roke** [reply, marked as answer]
> https://github.com/cloudbase/powershell-yaml/discussions/178#discussioncomment-11649453

Transferred to [`PowerShell/PowerShell/discussions/24701`](https://github.com/PowerShell/PowerShell/discussions/24701#discussion-7736596).

**Reply by beedell-roke** [reply]
> https://github.com/cloudbase/powershell-yaml/discussions/178#discussioncomment-11649645

@gabriel-samfira, per https://github.com/PowerShell/PowerShell/discussions/24701#discussioncomment-11652287, I'm beginning to wonder whether this might actually be related to `powershell-yaml`, and me reporting it here in my tired stupor was a stroke of good luck. Would you read the aforelinked "Crash 2" comment to see whether you think what I've stated has any merit?

Irrespective, since you asked in https://github.com/cloudbase/powershell-yaml/discussions/178#discussioncomment-11647451 and it now *might* be relevant, undermentioned is the environment information (that I ideally should have initially provided):

1.	~~~pwsh
	#!/usr/bin/env pwsh
	#Requires -PSEdition 'Core'
	#Requires -Version 7.5

	Get-InstalledModule -Name 'powershell-yaml' | ConvertTo-YAML
	~~~

1.	<blockquote>

	~~~YAML
	Name: powershell-yaml
	Version: 0.4.7
	Type: Module
	Description: Powershell module for serializing and deserializing YAML
	Author: Gabriel Adrian Samfira Alessandro Pilotti
	CompanyName: cloudbase
	Copyright: (c) 2016 Cloudbase Solutions SRL. All rights reserved.
	PublishedDate: 2023-03-02T13:18:18.0000000
	InstalledDate: 2024-10-19T20:39:11.7675540+01:00
	UpdatedDate:
	LicenseUri:
	ProjectUri:
	IconUri:
	Tags:
	- PSModule
	Includes:
	  Command: []
	  DscResource: []
	  Cmdlet: []
	  Function: []
	  RoleCapability: []
	  Workflow: []
	PowerShellGetFormatVersion:
	ReleaseNotes:
	Dependencies: []
	RepositorySourceLocation: https://www.powershellgallery.com/api/v2
	Repository: PSGallery
	PackageManagementProvider: NuGet
	AdditionalMetadata:
	  summary: Powershell module for serializing and deserializing YAML
	  installeddate: 2024-10-19T20:39:11.7675540+01:00
	  copyright: (c) 2016 Cloudbase Solutions SRL. All rights reserved.
	  Type: Module
	  PackageManagementProvider: NuGet
	  published: 2023-03-02 13:18:18
	  SourceName: PSGallery
	  tags: PSModule
	  description: Powershell module for serializing and deserializing YAML
	  InstalledLocation: C:\Users\rokej\OneDrive\Documents\PowerShell\Modules\powershell-yaml\0.4.7
	  IsPrerelease: false
	InstalledLocation: C:\Users\rokej\OneDrive\Documents\PowerShell\Modules\powershell-yaml\0.4.7
	~~~

	</blockquote>

**Reply by gabriel-samfira** [reply]
would you mind installing v0.4.11 and see if it still happens? Run: `Update-Module powershell-yaml`

**Reply by beedell-roke** [reply]
> https://github.com/cloudbase/powershell-yaml/discussions/178#discussioncomment-11653890

@gabriel-samfira, I don't appear to be able to automatically update to that version (at least, on Windows):

<blockquote>

~~~YAML
PS C:\Users\rokej> Update-Module -Name 'powershell-yaml'

Untrusted repository
You are installing the modules from an untrusted repository. If you trust this repository, change its
InstallationPolicy value by running the Set-PSRepository cmdlet. Are you sure you want to install the modules from
'PSGallery'?
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "N"): y
PS C:\Users\rokej> Get-InstalledModule -Name 'powershell-yaml' | Select-Object 'Version' | Format-List

Version : 0.4.7
~~~

</blockquote>

Has it definitely been published to PSGallery?

**Reply by gabriel-samfira** [reply]
yes: https://www.powershellgallery.com/packages/powershell-yaml/0.4.11

**Reply by gabriel-samfira** [reply]
try opening a new powershell session after the update and check the version again.

**Reply by beedell-roke** [reply]
> https://github.com/cloudbase/powershell-yaml/discussions/178#discussioncomment-11654164

@gabriel-samfira, thanks. That seems to have worked:

1. <details>

   1. ~~~pwsh
      exit
      ~~~
   1. ~~~pwsh
      pwsh
      ~~~

   </details>

1. 1. ~~~pwsh
      Update-Module -Name 'powershell-yaml' && `
      Get-InstalledModule -Name 'powershell-yaml' | Select-Object 'Version' | Format-List
      ~~~

   1. <blockquote>

      ~~~YAML
      Version : 0.4.11
      ~~~

      </blockquote>
